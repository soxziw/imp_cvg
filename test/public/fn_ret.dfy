method Negate(x: int) returns (r: int)
  requires x < 0
  ensures r >= 0
{
  var result : int;
  result := -x;
  return result;
}

// not verified
method Foo(x: int) returns (r: int)
  requires (x < 0)
{
  var y : int;
  y := x;
  y := Negate(y);
  assert y < 0;
  return 0;
}

// verified
method Bar(x: int) returns (r: int)
  requires (x < 0)
{
  var y : int;
  y := x;
  y := Negate(y);
  assert y >= 0;
  return 0;
}