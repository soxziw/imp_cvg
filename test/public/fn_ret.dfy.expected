Negate: verified
Foo: not verified
Bar: verified
